### DISCUSSION TITLE
First time contributor workflow approval

### DISCUSSION THREAD

**Discussion by jasonvarga** [community/community#14334] [Pull Requests]
When a first-time contributor makes a PR, we're required to push a button to allow them to run workflows. This thing:

![image](https://user-images.githubusercontent.com/105211/161783594-5e59d233-72db-4795-8ee7-ded6fc8f38a6.png)

This is fine, but if they push an additional commit, we have to push the button again.

Can we have the option to approve it once for the entirety of the PR?

**Comment by countvajhula** [comment]
+1 - This is exactly what I came here to suggest. The current options available on the Settings page are:

<img width="750" alt="Screen Shot 2022-06-14 at 12 37 10 PM" src="https://user-images.githubusercontent.com/401668/173674164-72f9961d-a28d-4717-aea7-c392eb9c14f8.png">

The first option seems too lax, while the second option seems right (to me), except that it is too strict in the way it currently works, as the above commenter pointed out.

In practice, many pull requests come from first-time contributors, and saying "approve and run" the first time indicates that we think the contribution is legitimate. It would be nice if approving it just once meant that subsequent commits by the contributor (which may be numerous, and may happen over a long period while the pull request is in progress and goes through review stages) would be run without requiring action each time. This would make it a more pleasant experience for such contributors since they would be able to leverage the testing infrastructure of a new repo they've never contributed to, without the maintainer having to approve everything.
